### PR TITLE
Mark -ingester.limit-inflight-requests-using-grpc-method-limiter and -distributor.limit-inflight-requests-using-grpc-method-limiter as stable and enable it by default

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

`-ingester.limit-inflight-requests-using-grpc-method-limiter` and `-distributor.limit-inflight-requests-using-grpc-method-limiter` were introduced in Mimir 2.11, disabled by default. Over the time we enabled it in all Mimir clusters at Grafana Labs and we haven't seen any issue (on the contrary, we've seen benefits). In this PR I propose to enable it by default. However, since these are experimental flags and experimental stuff shouldn't be enabled by default, I'm also marking the feature as stable. Finally, unless over the course of the next 2 releases we'll find any issue / bug, I think the config option to disable it should be removed in the long term, so I'm contextually marking the flags as deprecated so that we can remove them in Mimir 2.14.

Note to reviewers:
- I changed some tests to run both with and without gRPC limiter enabled
  - In case gRPC limiter is enabled and the request is rejected by `StartPushRequest()`, the error doesn't need to implement `middleware.OptionalLogging` or being a gRPC error because (a) the gRPC handler never log such errors and (b) the gRPC handler always returns `codes.Unavailable` which is then translated to `http.StatusInternalServerError`
  - I recommend to review these changes with "hide whitespace changes" enabled

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
Makes sense, we want to get rid of those flags.

File: pkg/distributor/distributor.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ type Config struct {
 	PushWrappers []PushWrapper `yaml:"-"`
 
 	WriteRequestsBufferPoolingEnabled           bool `yaml:"write_requests_buffer_pooling_enabled" category:"experimental"`
-	LimitInflightRequestsUsingGrpcMethodLimiter bool `yaml:"limit_inflight_requests_using_grpc_method_limiter" category:"experimental"`
+	LimitInflightRequestsUsingGrpcMethodLimiter bool `yaml:"limit_inflight_requests_using_grpc_method_limiter" category:"advanced"` // TODO Remove the configuration option in Mimir 2.14, keeping the same behavior as if it's enabled
```
Comment (pstibrany) [inline pkg/distributor/distributor.go:209]: Let's use `category:"deprecated"`.

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -172,7 +172,7 @@ type Config struct {
 	ReadPathMemoryUtilizationLimit       uint64  `yaml:"read_path_memory_utilization_limit" category:"experimental"`
 	LogUtilizationBasedLimiterCPUSamples bool    `yaml:"log_utilization_based_limiter_cpu_samples" category:"experimental"`
 
-	LimitInflightRequestsUsingGrpcMethodLimiter bool `yaml:"limit_inflight_requests_using_grpc_method_limiter" category:"experimental"`
+	LimitInflightRequestsUsingGrpcMethodLimiter bool `yaml:"limit_inflight_requests_using_grpc_method_limiter" category:"advanced"` // TODO Remove the configuration option in Mimir 2.14, keeping the same behavior as if it's enabled.
```
Comment (pstibrany) [inline pkg/ingester/ingester.go:175]: suggestion: `category:"deprecated`

**Review by pstibrany — APPROVED** [review]
Thank you!

File: pkg/ingester/ingester.go (hunk 2 above)
Comment (pracucci) [reply pkg/ingester/ingester.go]: 🤦  Yes, sure https://github.com/grafana/mimir/pull/7360/commits/2da2f68495b839ff963fa4a97321475059b5a488

File: pkg/distributor/distributor.go (hunk 1 above)
Comment (pracucci) [reply pkg/distributor/distributor.go]: 🤦  Yes, sure https://github.com/grafana/mimir/pull/7360/commits/2da2f68495b839ff963fa4a97321475059b5a488

**Pull request merged**